extensions/amp-sticky-ad/0.1/amp-sticky-ad.js

```diff
@@ -28,15 +30,71 @@ class AmpStickyAd extends AMP.BaseElement {
   isLayoutSupported(layout) {
     return layout == Layout.NODISPLAY;
   }
-
-  /** @override */
+  /**
+   * @override
+   * @this {undefined}  // Make linter happy
+   */
   buildCallback() {
     /** @const @private {boolean} */
     this.isExperimentOn_ = isExperimentOn(this.getWin(), TAG);
     if (!this.isExperimentOn_) {
       dev.warn(TAG, `TAG ${TAG} disabled`);
       return;
     }
+    this.viewport_ = this.getViewport();
+    this.isDisplayed_ = false;
+    this.initialScrollTop_ = this.viewport_.getScrollTop();
+    this.scrollHeight_ = this.viewport_.getScrollHeight();
+    this.viewportHeight_ = this.viewport_.getSize().height;
+    /** @const @private {!Vsync} */
+    this.vsync_ = vsyncFor(this.getWin());
+    this.viewport_.onScroll(() => {
+      if (!this.isDisplayed_) {
+        const scrollDist =
+            (this.viewport_.getScrollTop() - this.initialScrollTop_);
+        if (this.viewportHeight_ < Math.abs(scrollDist)) {
+          // scroll down
+          if (scrollDist < 0) {
+            if (this.viewport_.getScrollTop() < this.viewportHeight_) {
+              // Not sure about this
+              this.initialScrollTop_ = this.viewport_.getScrollTop();
+              return;
+            }
+          } /*scroll up*/else {
```

Comment:
This comment should go on a new line after `} else {` as `// Scroll up` or such.